### PR TITLE
Refactor module sync control

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Refactor ModuleTemplate synchronization logic according to the #542 
- Adjust the test cases

**Related issue(s)**
See also: #542

### CODE REVIEW

File: api/v1beta1/kyma_types.go [hunk 1]
```diff
@@ -420,3 +422,28 @@ func (kyma *Kyma) AllModulesReady() bool {
 	}
 	return true
 }
+
+func (kyma *Kyma) SyncEnabled() bool {
+	syncVal, found := kyma.GetLabels()[SyncLabel]
```
Comment (ruanxin) [inline api/v1beta1/kyma_types.go:427]: 1. the labels might be nil, nil pointer should be detected first.
2. in this PR, it's better the value of `kyma.sync` field still respect.

**Comment by Tomasz-Smelcerz-SAP** [comment]
Manual tests performed:
```
========================================
Testing single module sync (no beta/internal)
========================================

1) Create kyma and moduletemplate without 'sync' labels
   - should sync
   * OK

2) Create kyma and moduletemplate with empty 'sync' labels
   - should sync
   * OK

3) Create kyma with 'sync=false' and moduletemplate with 'sync=true' labels
   - should not sync
   * OK

4) Create kyma with 'sync=true' and moduletemplate with 'sync=false' labels
   - should not sync
   * OK

5) Create kyma with 'sync=true' and moduletemplate with 'sync=true' labels
   - should sync
   * OK

========================================
Testing Beta/Internal feature
========================================

1) Create standard ModuleTemplate(sync enabled) and a beta one(sync enabled)
   Enable sync on Kyma object
   - only standard module should sync
   * OK

2) Like 1) but in addition enable internal on the Kyma
   - only standard module should sync
   * OK

3) Like 1) but in addition enable beta on the Kyma
   - both modules should sync
   * OK


4) Create standard ModuleTemplate(sync enabled) and an internal one(sync enabled)
   Enable sync on Kyma object
   - only standard module should sync
   * OK

5) Like 4) but in addition enable beta on the Kyma
   - only standard module should sync
   * OK

6) Like 4) but in addition enable internal on the Kyma
   - both modules should sync
   * OK

```

**Review by ruanxin — APPROVED** [review]

**Comment by ruanxin** [comment]
the change is included in https://github.com/kyma-project/lifecycle-manager/pull/553

**Pull request closed without merging**